### PR TITLE
[SDL-0188] Fix resume 2nd app in case of different module types

### PR DESCRIPTION
**Pull request by SNiukalov**
Fixes #[JIRA-7383](https://adc.luxoft.com/jira/browse/FORDTCN-7383)

This PR is **ready** for review.

### Summary
Fixed behavior for freeze and repeat request after successful response.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_pending_resumption_handler.cc [hunk 1]
```diff
@@ -220,6 +226,19 @@ bool RCPendingResumptionHandler::IsPendingForResponse(
   return it != waiting_for_response_modules_.end();
 }
 
+bool RCPendingResumptionHandler::IsAnotherAppsSubscribedOnTheSameModule(
+    const ModuleUid& module_uid) const {
+  const auto& it = subscribed_modules.find(module_uid);
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_pending_resumption_handler.cc:231]: We are doing subscriptions not only via RCPendingResumptionHandler.
subscribed_modules it is only subscribed via resumption. But at the same time, regular subscriptions are possible. 

Please reimplement IsAnotherAppsSubscribedOnTheSameModule to check the status subscription of the apps but not only resumed one

**Review by LuxoftAKutsan — CHANGES_REQUESTED** [review]

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_pending_resumption_handler.cc (hunk 1 above)
Comment (SNiukalov) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_pending_resumption_handler.cc]: @LuxoftAKutsan 
Fixed in:https://github.com/LuxoftSDL/sdl_core/pull/119/commits/e8ebdba4bca068592d98be4cdcee87e4b608ae29

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_pending_resumption_handler.h [hunk 2]
```diff
@@ -20,6 +19,8 @@ namespace rc_rpc_plugin {
 class RCPendingResumptionHandler
     : public resumption::ExtensionPendingResumptionHandler {
  public:
+  typedef std::map<int32_t, smart_objects::SmartObject> PendingRequestList;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_pending_resumption_handler.h:22]: Move typedef closer to variable with it

**Review by LuxoftAKutsan — APPROVED** [review]

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_pending_resumption_handler.h (hunk 2 above)
Comment (SNiukalov) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_pending_resumption_handler.h]: @LuxoftAKutsan 
Fixed in:https://github.com/LuxoftSDL/sdl_core/pull/119/commits/1c3cb71ed51e588856fa14d5868096600d87bf0a

**Pull request merged**